src/documentation/ideas/content.ts

```diff
@@ -14,25 +14,29 @@ const ideasQuery = (languageId: string): string => {
     throw new Error("Invalid language id.");
   }
   return `
-    *[_type == "pythonIdea" && language == "${languageId}" && !(_id in path("drafts.**"))]{
-      _id, name, language, compatibility, image, slug,
-      content[] {
-        ...,
-        markDefs[]{
+    *[_type == "pythonIdeasConfig" && language == "${languageId}" && !(_id in path("drafts.**"))][0]{
+      pythonIdeasOrder[]->{
+        _id, name, language, compatibility, image, slug,
+        content[] {
           ...,
-          _type == "toolkitInternalLink" => {
-              "slug": @.reference->slug,
-              "targetType": @.reference->_type
+          markDefs[]{
+            ...,
+            _type == "toolkitInternalLink" => {
+                "slug": @.reference->slug,
+                "targetType": @.reference->_type
+            }
           }
-        }
-      },
+        },
+      }
     }`;
 };
 
 const adaptContent = (result: any): Idea[] | undefined => {
-  const ideas = result as Idea[];
-  if (ideas.length === 0) {
-    return undefined;
+  const ideas = result?.pythonIdeasOrder as Idea[];
+  if (ideas) {
+    if (ideas.length === 0) {
```

Comment:
This looks slightly odd.

Reply:
More logical check here -> [`80ad3b2`](https://github.com/microbit-foundation/python-editor-next/pull/837/commits/80ad3b25970c049196b391afded3b030e5e20292).